sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -156,9 +157,16 @@ object JDBCRDD extends Logging {
     val url = options.url
     val dialect = JdbcDialects.get(url)
     val quotedColumns = requiredColumns.map(colName => dialect.quoteIdentifier(colName))
+
+    val conn = JdbcUtils.createConnectionFactory(options)
+    options.preActions match {
+      case Some(i) =>
+        runQuery(conn(), i, options)
```

Comment:
nit: `conn()` -> `conn` ?

Reply:
`scanTable` returns `JDBCRDD` with `JdbcUtils.createConnectionFactory(options)`, not with `JdbcUtils.createConnectionFactory(options)()`.

BTW `scanTable` is called per RDD so I moved the implementation to `resolveTable`. Here, I used `conn` as the argument for `runQuery`.

Reply:
Does this call `conn` have a side-effect? According to [this document](https://github.com/databricks/scala-style-guide#parentheses), `()` means that it has a side-effect.

IMHO, since other places use `conn`, I guess that this call does not have a side-effect.

Reply:
Thanks for pointing the reference. It makes sense.
BTW, in the current diff, all the `runQuery` calls have `conn` instead of `conn()`.

Reply:
Sure. Now, they are fine.